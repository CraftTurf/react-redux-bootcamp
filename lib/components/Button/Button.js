'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactCssModules = require('react-css-modules');

var _reactCssModules2 = _interopRequireDefault(_reactCssModules);

var _nodeNoop = require('node-noop');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Button = require('./Button.scss');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = function Button(_ref) {
  var _classNames;

  var onClick = _ref.onClick,
      children = _ref.children,
      disabled = _ref.disabled,
      styleSheet = _ref.styleSheet,
      tabIndex = _ref.tabIndex,
      type = _ref.type;
  return _react2.default.createElement(
    'button',
    {
      onClick: onClick,
      disabled: disabled,
      tabIndex: tabIndex,
      styleName: 'button',
      className: (0, _classnames2.default)(styleSheet.button, (_classNames = {}, (0, _defineProperty3.default)(_classNames, _Button2.default.disabled, disabled), (0, _defineProperty3.default)(_classNames, styleSheet.disabled, disabled), _classNames)),
      type: type
    },
    children
  );
};


Button.defaultProps = {
  onClick: _nodeNoop.noop,
  children: 'Click me',
  disabled: false,
  tabIndex: 0,
  styleSheet: {
    button: 'button'
  },
  type: 'button'
};

exports.default = (0, _reactCssModules2.default)(Button, _Button2.default);