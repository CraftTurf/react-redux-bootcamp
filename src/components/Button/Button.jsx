// @flow
import React from 'react';
import type { Node as ReactNode } from 'react';
import cssModule from 'react-css-modules';
import { noop } from 'node-noop';
import classNames from 'classnames';

import s from './Button.scss';

export type ButtonProps = {
  onClick: SyntheticEvent<HTMLButtonElement>,
  children: ReactNode,
  disabled: boolean,
  tabIndex: number,
  type: "button" | "submit" | "reset",
  styleSheet: {
    button?: string,
    disabled?: string,
  }
};

const Button = ({
  onClick,
  children,
  disabled,
  styleSheet,
  tabIndex,
  type,
}: ButtonProps): ReactNode => (
  <button
    onClick={onClick}
    disabled={disabled}
    tabIndex={tabIndex}
    styleName="button"
    className={classNames(styleSheet.button, { [s.disabled]: disabled, [styleSheet.disabled]: disabled })}
    type={type}
  >
    {children}
  </button>
);


Button.defaultProps = {
  onClick: noop,
  children: 'Click me',
  disabled: false,
  tabIndex: 0,
  styleSheet: {
    button: 'button'
  },
  type: 'button'
};

export default cssModule(Button, s);
