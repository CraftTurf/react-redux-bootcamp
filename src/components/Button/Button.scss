@import '../../style/palette';
@import '../../style/typography';
@import '../../style/grid';
@import '../../style/normalize/variables';

.button,
.disabled {
  @include fs-bold(2);

  padding: $fg-gutter;
  background-color: c-color(funfun);
  color: c-color(dudu);
  border: 1px solid c-color(dudu);
  border-radius: 0;
}

.disabled {
  background-color: c-color(funfun, .1);
  color: c-color(dudu, .25);
  border: 1px solid c-color(dudu, .25);
}
