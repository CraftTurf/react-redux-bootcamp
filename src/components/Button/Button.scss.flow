// @flow
/* This file is automatically generated by css-modules-flow-types */
declare module.exports: {|
  +'button': string;
  +'col-lg': string;
  +'col-lg-1': string;
  +'col-lg-10': string;
  +'col-lg-11': string;
  +'col-lg-12': string;
  +'col-lg-2': string;
  +'col-lg-3': string;
  +'col-lg-4': string;
  +'col-lg-5': string;
  +'col-lg-6': string;
  +'col-lg-7': string;
  +'col-lg-8': string;
  +'col-lg-9': string;
  +'col-md': string;
  +'col-md-1': string;
  +'col-md-10': string;
  +'col-md-11': string;
  +'col-md-12': string;
  +'col-md-2': string;
  +'col-md-3': string;
  +'col-md-4': string;
  +'col-md-5': string;
  +'col-md-6': string;
  +'col-md-7': string;
  +'col-md-8': string;
  +'col-md-9': string;
  +'col-sm': string;
  +'col-sm-1': string;
  +'col-sm-10': string;
  +'col-sm-11': string;
  +'col-sm-12': string;
  +'col-sm-2': string;
  +'col-sm-3': string;
  +'col-sm-4': string;
  +'col-sm-5': string;
  +'col-sm-6': string;
  +'col-sm-7': string;
  +'col-sm-8': string;
  +'col-sm-9': string;
  +'col-xs': string;
  +'col-xs-1': string;
  +'col-xs-10': string;
  +'col-xs-11': string;
  +'col-xs-12': string;
  +'col-xs-2': string;
  +'col-xs-3': string;
  +'col-xs-4': string;
  +'col-xs-5': string;
  +'col-xs-6': string;
  +'col-xs-7': string;
  +'col-xs-8': string;
  +'col-xs-9': string;
  +'disabled': string;
  +'f-bold-1': string;
  +'f-bold-2': string;
  +'f-bold-3': string;
  +'f-bold-4': string;
  +'f-bold-5': string;
  +'f-bold-6': string;
  +'f-light-1': string;
  +'f-light-2': string;
  +'f-light-3': string;
  +'f-light-4': string;
  +'f-light-5': string;
  +'f-light-6': string;
  +'f-regular-1': string;
  +'f-regular-2': string;
  +'f-regular-3': string;
  +'f-regular-4': string;
  +'f-regular-5': string;
  +'f-regular-6': string;
  +'off-lg-1': string;
  +'off-lg-10': string;
  +'off-lg-11': string;
  +'off-lg-2': string;
  +'off-lg-3': string;
  +'off-lg-4': string;
  +'off-lg-5': string;
  +'off-lg-6': string;
  +'off-lg-7': string;
  +'off-lg-8': string;
  +'off-lg-9': string;
  +'off-lg-reset': string;
  +'off-md-1': string;
  +'off-md-10': string;
  +'off-md-11': string;
  +'off-md-2': string;
  +'off-md-3': string;
  +'off-md-4': string;
  +'off-md-5': string;
  +'off-md-6': string;
  +'off-md-7': string;
  +'off-md-8': string;
  +'off-md-9': string;
  +'off-md-reset': string;
  +'off-sm-1': string;
  +'off-sm-10': string;
  +'off-sm-11': string;
  +'off-sm-2': string;
  +'off-sm-3': string;
  +'off-sm-4': string;
  +'off-sm-5': string;
  +'off-sm-6': string;
  +'off-sm-7': string;
  +'off-sm-8': string;
  +'off-sm-9': string;
  +'off-sm-reset': string;
  +'off-xs-1': string;
  +'off-xs-10': string;
  +'off-xs-11': string;
  +'off-xs-2': string;
  +'off-xs-3': string;
  +'off-xs-4': string;
  +'off-xs-5': string;
  +'off-xs-6': string;
  +'off-xs-7': string;
  +'off-xs-8': string;
  +'off-xs-9': string;
  +'off-xs-reset': string;
  +'row': string;
|};
