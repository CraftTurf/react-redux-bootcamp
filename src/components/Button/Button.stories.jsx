import React from 'react';
import { storiesOf } from '@storybook/react';
import { checkA11y } from '@storybook/addon-a11y';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';

import Button from './Button';

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addDecorator(checkA11y)
  .add('default', () =>
    <Button onClick={action('clicked')}>
      Hello
    </Button>
  )
  .add('disabled', () =>
    <Button onClick={action('button clicked')} disabled>
      Hello
    </Button>
  )
  .add('with knobs', () =>
    <Button onClick={action('button clicked')}>
      {text('Label', 'Hello')}
    </Button>
  )
  .add('link button', () =>
    <Button onClick={linkTo('Button', 'some emojies as the text')}>
      Next Story
    </Button>
  )
  .add('some emojies as the text', () =>
    <Button onClick={action('button clicked')}>😀 😎 👍 💯</Button>
  );
