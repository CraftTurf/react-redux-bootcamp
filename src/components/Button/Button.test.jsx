import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import Button from './Button';

enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <Button />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('onClick', () => {
    let wrapper;
    let onClick;

    beforeEach(() => {
      onClick = jest.fn();
      wrapper = shallow(<Button onClick={onClick} />);
    });

    it('should have an onClick property', () => {
      expect(wrapper.props().onClick).toBeDefined();
    });

    it('should react to a click', () => {
      wrapper.simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
