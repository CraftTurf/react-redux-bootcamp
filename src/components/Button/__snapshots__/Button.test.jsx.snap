exports[`<Button /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={false}
    onClick={[Function]}
    styleSheet={
        Object {
            "button": "button",
          }
    }
    tabIndex={0}
    type="button">
    Click me
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me",
      "className": "button button",
      "disabled": false,
      "onClick": [Function],
      "tabIndex": 0,
      "type": "button",
    },
    "ref": null,
    "rendered": "Click me",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me",
        "className": "button button",
        "disabled": false,
        "onClick": [Function],
        "tabIndex": 0,
        "type": "button",
      },
      "ref": null,
      "rendered": "Click me",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
